source/javascripts/hooks/api/useFetchCallback.ts

```diff
@@ -25,55 +25,59 @@ function useFetchCallback<T, E>(
 	const abortController = useRef<AbortController>();
 	const throwError = useAsyncError();
 
-	const call = useCallback(() => {
-		setStatusCode(undefined);
-		setFailed(undefined);
+	const call = useCallback(
+		(options?: RequestInit) => {
+			setStatusCode(undefined);
+			setFailed(undefined);
 
-		abortController.current?.abort();
-		abortController.current = new AbortController();
+			abortController.current?.abort();
+			abortController.current = new AbortController();
 
-		(async () => {
-			setLoading(true);
-			try {
-				if (url === "") {
-					throwError(new Error("Url is required"));
-					return;
-				}
-
-				const result = await fetch(url, {
-					signal: abortController.current?.signal,
-					...init,
-					headers: {
-						"Content-Type": "application/json",
-						Accept: "application/json",
-						...init?.headers
+			(async () => {
+				setLoading(true);
+				try {
+					if (url === "") {
+						throwError(new Error("Url is required"));
+						return;
 					}
-				});
 
-				setStatusCode(result.status);
+					const result = await fetch(url, {
+						signal: abortController.current?.signal,
+						...init,
+						...options,
```

Comment:
FYI This options line is the only relevant change, all the other stuff is just formatting. Allowing additional options at this point means we only need to do the yml formatting when we actually want to call the endpoint. 

`useFetchCallback` previously only had the option to do this with the init object. The previous implementation did the json stringify on each render.

<img width="543" alt="Screenshot 2023-01-13 at 13 56 10" src="https://user-images.githubusercontent.com/107245372/212325269-b07ea4cf-fd31-43f0-a60a-1a44d1a9d629.png">

This was kind of passable but not ideal but for the yml transform it would be even worse.